lib/augmentConfig.js

```diff
@@ -370,4 +404,53 @@ function addProcessorFunctions(config) {
 	return config;
 }
 
-module.exports = { augmentConfigExtended, augmentConfigFull };
+/**
+ * @param {StylelintConfig} fullConfig
+ * @param {string} configDir
+ * @param {string} filePath
+ * @return {StylelintConfig}
+ */
+function applyOverrides(fullConfig, configDir, filePath) {
+	let { overrides, ...config } = fullConfig;
+
+	if (!overrides) {
+		return config;
+	}
+
+	if (!Array.isArray(overrides)) {
+		throw Error(
+			'The `overrides` configuration property should be an array, e.g. { "overrides": [{ "files": "*.css", "rules": {}}] }.',
+		);
+	}
+
+	for (const override of overrides) {
+		const { files, ...configOverrides } = override;
+
+		if (!files) {
+			throw Error(
+				'Every object in the `overrides` configuration property should have a `files` property with globs, e.g. { "overrides": [{ "files": "*.css", "rules": {}}] }.',
+			);
+		}
+
+		let filesGlobs = Array.isArray(files) ? files : [files];
+
+		filesGlobs = filesGlobs
+			.map((glob) => {
+				if (path.isAbsolute(glob.replace(/^!/, ''))) {
+					return glob;
+				}
+
+				return globjoin(configDir, glob);
+			})
+			// Glob patterns for micromatch should be in POSIX-style
+			.map(slash);
+
+		if (micromatch.isMatch(filePath, filesGlobs)) {
+			config = mergeConfigs(config, configOverrides);
+		}
+	}
+
+	return config;
+}
```

Comment:
[question] Does it make sense to check the `rules` key existence in addition to `files`?

For example:

```js
if (!rules) {
	throw Error(
		'Every object in the `overrides` configuration property should have a `rules` property ...',
	);
}
```

Reply:
It doesn't necessarily need a `rules` key as the user could override something else, e.g. `customSyntax` when linting both CSS and SCSS with `stylelint "**/*.{css,scss}"`:

```json
{
  "overrides": [{
    "files": "*.scss",
    "customSyntax": "postcss-scss"
  }],
  "rules": {
    ..
  }
}
```

We could check that there's at least one other key than `files`, though.

Reply:
@jeddy3 I see. Thank you!